pkg/hubble/filters/tcp.go

```diff
@@ -13,7 +13,11 @@ import (
 
 func filterByTCPFlags(flags []*flowpb.TCPFlags) (FilterFunc, error) {
 	return func(ev *v1.Event) bool {
-		l4tcp := ev.GetFlow().GetL4().GetTCP()
+		flow := ev.GetFlow()
+		if flow == nil || flow.Type != flowpb.FlowType_L3_L4 {
```

Comment:
I have an alternative proposal: Instead of checking if this is an L3/L4 flow, let's just check if `tl4tcp.GetFlags()` is nil. i.e. add an `if flowFlags == nil { return false }` here:

https://github.com/cilium/cilium/blob/76d34dbdfc1bf14cb8976a86ccf1e98dae6d72c5/pkg/hubble/filters/tcp.go#L20

Otherwise the bug might be re-introduced if we ever change the L3/L4 parser to not always populate the TCP flags.

Reply:
Okay, I have changed the logic as suggested.